src/audio/selector/selector_generic.c

```diff
@@ -254,6 +254,10 @@ static void process_frame_s32le(int32_t dst[], int dst_channels,
 	int64_t accum;
 	int i, j;
 
+	if (dst_channels > SEL_SINK_CHANNELS_MAX ||
+	    src_channels > SEL_SOURCE_CHANNELS_MAX)
+		return;
```

Comment:
I actually wonder if this is what should be done. I'd consider returning an error or clamping the value to the maximum allowed one here.

Reply:
@lyakh is this aligning with the other frame formats in selector or should this be done by the caller ?

Reply:
this functions returns `void`

Reply:
@lyakh I mean do it here in the caller. This way we can check all users.
```c
static int selector_process(struct processing_module *mod,
			    struct input_stream_buffer *input_buffers,
			    int num_input_buffers,
			    struct output_stream_buffer *output_buffers,
			    int num_output_buffers)
{
	struct comp_data *cd = module_get_private_data(mod);
	uint32_t avail_frames = input_buffers[0].size;

	comp_dbg(mod->dev, "selector_process()");

	if (!avail_frames)
		return PPL_STATUS_PATH_STOP;

	/* copy selected channels from in to out */
	cd->sel_func(mod, input_buffers, output_buffers, avail_frames);

	return 0;
}
```

Reply:
Its a cheap check here regardless of single/multi channel and static analysis can be closed for this one.